From 100d4711e808ce2205d22ad8dad56b36e49f0319 Mon Sep 17 00:00:00 2001
From: Sebastian Huang <sebastian_huang@accton.com>
Date: Mon, 1 Dec 2025 15:53:38 +0800
Subject: [PATCH] base-files: ipcalc.sh: revert IP range check error to warning

Allow the LAN IP to be inside of the DHCP range, only show the warning message.
---
 package/base-files/files/bin/ipcalc.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/package/base-files/files/bin/ipcalc.sh b/package/base-files/files/bin/ipcalc.sh
index 871a49ed6e..6a8cf67ec4 100755
--- a/package/base-files/files/bin/ipcalc.sh
+++ b/package/base-files/files/bin/ipcalc.sh
@@ -133,8 +133,9 @@ _ip2str START "$start"
 _ip2str END "$end"
 
 if [ "$start" -le "$ipaddr" ] && [ "$ipaddr" -le "$end" ]; then
-    echo "error: address $IP inside range $START..$END" >&2
-    exit 1
+    echo "warning: ipaddr inside range - this might not be supported in future releases of Openwrt" >&2
+    # echo "error: address $IP inside range $START..$END" >&2
+    # exit 1
 fi
 
 echo "START=$START"
-- 
2.34.1

